### PR TITLE
Add support for Node 20

### PR DESCRIPTION
**Pull request by hossam-nasr**
Related to https://github.com/Azure/azure-functions-nodejs-library/issues/170

node20: 1.61GB
<details>
<summary>dive results</summary>

image

image

</details>

node20-slim: 1.28GB

<details>
<summary>dive results</summary>

image

image

</details>

node20-appservice: 1.42GB

<details>
<summary>dive results</summary>

image

image

</details>

### PR information

- [x] The title of the PR is clear and informative.
- [x] There are a small number of commits, each of which has an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/development-docs/cleaning-up-commits.md).
- [x] If applicable, the PR references the bug/issue that it fixes in the description.
- [] New Unit tests were added for the changes made and CI is passing.

### CODE REVIEW

**Comment by CooperLink** [comment]
Looks pretty good please just make sure you have updated every pipeline. It looks like republish got skipped rn

**Comment by hossam-nasr** [comment]
@CooperLink Done!

**Review by ejizba — DISMISSED** [review]

File: host/4/publish-appservice-stage.yml [hunk 1]
```diff
@@ -156,6 +156,20 @@ steps:
       docker system prune -a -f
     displayName: tag and push node images
     continueOnError: false
+  
+  - bash: |
+      set -e
+      docker pull $SOURCE_REGISTRY/node:$(PrivateVersion)-node20-appservice
+
+      docker tag $SOURCE_REGISTRY/node:$(PrivateVersion)-node20-appservice $TARGET_REGISTRY/node:$(TargetVersion)-appservice-stage$(StageNumber)
```
Comment (CooperLink) [inline host/4/publish-appservice-stage.yml]: This sets the default node version to node 20 which is bad practice unless totally intended.  Could upgrade some small subset of customers randomly to this version. Remove

File: host/4/publish-appservice-stage.yml [hunk 2]
```diff
@@ -156,6 +156,20 @@ steps:
...
+      docker tag $SOURCE_REGISTRY/node:$(PrivateVersion)-node20-appservice $TARGET_REGISTRY/node:$(TargetVersion)-node20-appservice-stage$(StageNumber)
+
+      docker push $TARGET_REGISTRY/node:$(TargetVersion)-appservice-stage$(StageNumber)
```
Comment (CooperLink) [inline host/4/publish-appservice-stage.yml]: same

File: host/4/publish.yml [hunk 3]
```diff
@@ -325,6 +325,38 @@ steps:
...
     displayName: tag and push node18 images
     continueOnError: false
+  
+  - bash: |
+      set -e
+      docker pull $SOURCE_REGISTRY/node:$(PrivateVersion)-node20
+      docker pull $SOURCE_REGISTRY/node:$(PrivateVersion)-node20-slim
+      docker pull $SOURCE_REGISTRY/node:$(PrivateVersion)-node20-appservice
+
+      docker tag $SOURCE_REGISTRY/node:$(PrivateVersion)-node20 $TARGET_REGISTRY/node:$(TargetVersion)
```
Comment (CooperLink) [inline host/4/publish.yml]: Remove this set of tags.

File: host/4/publish.yml [hunk 4]
```diff
@@ -325,6 +325,38 @@ steps:
...
+      docker tag $SOURCE_REGISTRY/node:$(PrivateVersion)-node20-appservice $TARGET_REGISTRY/node:$(TargetVersion)-appservice
+      docker tag $SOURCE_REGISTRY/node:$(PrivateVersion)-node20-appservice $TARGET_REGISTRY/node:$(TargetVersion)-appservice-quickstart
+
+      docker tag $SOURCE_REGISTRY/node:$(PrivateVersion)-node20 $TARGET_REGISTRY/node:$(TargetVersion)-node20
+      docker tag $SOURCE_REGISTRY/node:$(PrivateVersion)-node20-slim $TARGET_REGISTRY/node:$(TargetVersion)-node20-slim
+      docker tag $SOURCE_REGISTRY/node:$(PrivateVersion)-node20-appservice $TARGET_REGISTRY/node:$(TargetVersion)-node20-appservice
+      docker tag $SOURCE_REGISTRY/node:$(PrivateVersion)-node20-appservice $TARGET_REGISTRY/node:$(TargetVersion)-node20-appservice-quickstart
+
+
+      docker push $TARGET_REGISTRY/node:$(TargetVersion)
```
Comment (CooperLink) [inline host/4/publish.yml]: Remove this set of pushes.

File: host/4/republish.yml [hunk 5]
```diff
@@ -264,6 +264,33 @@ steps:
...
+
+      docker tag $SOURCE_REGISTRY/node:$(PrivateVersion)-node20 $TARGET_REGISTRY/node:$(TargetVersion)
+      docker tag $SOURCE_REGISTRY/node:$(PrivateVersion)-node20-slim $TARGET_REGISTRY/node:$(TargetVersion)-slim
+      docker tag $SOURCE_REGISTRY/node:$(PrivateVersion)-node20-appservice $TARGET_REGISTRY/node:$(TargetVersion)-appservice
+
+      docker tag $SOURCE_REGISTRY/node:$(PrivateVersion)-node20 $TARGET_REGISTRY/node:$(TargetVersion)-node20
+      docker tag $SOURCE_REGISTRY/node:$(PrivateVersion)-node20-slim $TARGET_REGISTRY/node:$(TargetVersion)-node20-slim
+      docker tag $SOURCE_REGISTRY/node:$(PrivateVersion)-node20-appservice $TARGET_REGISTRY/node:$(TargetVersion)-node20-appservice
+
+      docker push $TARGET_REGISTRY/node:$(TargetVersion)
```
Comment (CooperLink) [inline host/4/republish.yml]: rm

File: host/4/republish.yml [hunk 6]
```diff
@@ -264,6 +264,33 @@ steps:
     displayName: tag and push node18 images
     continueOnError: false
 
+  - bash: |
+      set -e
+      docker pull $SOURCE_REGISTRY/node:$(PrivateVersion)-node20
+      docker pull $SOURCE_REGISTRY/node:$(PrivateVersion)-node20-slim
+      docker pull $SOURCE_REGISTRY/node:$(PrivateVersion)-node20-appservice
+
+      docker tag $SOURCE_REGISTRY/node:$(PrivateVersion)-node20 $TARGET_REGISTRY/node:$(TargetVersion)
```
Comment (CooperLink) [inline host/4/republish.yml]: rm

File: host/4/publish-appservice-stage-sovereign-clouds.yml [hunk 7]
```diff
@@ -165,6 +165,20 @@ steps:
           docker system prune -a -f
         displayName: tag and push node images for stage ${{stage}}
         continueOnError: false
+      
+      - bash: |
+          set -e
+          docker pull $SOURCE_REGISTRY/node:$(PrivateVersion)-node20-appservice
+          
+          docker tag $SOURCE_REGISTRY/node:$(PrivateVersion)-node20-appservice $TARGET_REGISTRY/node:$(TargetVersion)-appservice-$(CloudName)-stage${{stage}}
```
Comment (CooperLink) [inline host/4/publish-appservice-stage-sovereign-clouds.yml]: remove

File: host/4/publish-appservice-stage-sovereign-clouds.yml [hunk 8]
```diff
@@ -165,6 +165,20 @@ steps:
...
         continueOnError: false
+      
+      - bash: |
+          set -e
+          docker pull $SOURCE_REGISTRY/node:$(PrivateVersion)-node20-appservice
+          
+          docker tag $SOURCE_REGISTRY/node:$(PrivateVersion)-node20-appservice $TARGET_REGISTRY/node:$(TargetVersion)-appservice-$(CloudName)-stage${{stage}}
+          docker tag $SOURCE_REGISTRY/node:$(PrivateVersion)-node20-appservice $TARGET_REGISTRY/node:$(TargetVersion)-node20-appservice-$(CloudName)-stage${{stage}}
+          
+          docker push $TARGET_REGISTRY/node:$(TargetVersion)-appservice-$(CloudName)-stage${{stage}}
```
Comment (CooperLink) [inline host/4/publish-appservice-stage-sovereign-clouds.yml]: rm

**Review by CooperLink — CHANGES_REQUESTED** [review]
We don't want to set node 20 as teh default node version atleast not yet.  Also those image sizes are very big. But, I don't have any reason to believe that anything was done incorrectly.  It all looks good to me. We just need to discuss what precaching an image like that will mean

File: host/4/publish-appservice-stage.yml (hunk 1 above)
Comment (hossam-nasr) [reply host/4/publish-appservice-stage.yml]: Not intended, removed

**Comment by hossam-nasr** [comment]
Hi @CooperLink, thanks for the review! Defaulting to node 20 was not intended and I have removed it. For the image sizes, looking at dive results compared to Node 18's in Eric's [PR](https://github.com/Azure/azure-functions-docker/pull/706), it appears that the main difference is in extension bundles size increase. I tried building node 18 images today with the updated host and extension bundle versions, and I got very similar results to these node 20 images

**Review by CooperLink — APPROVED** [review]
This looks good to me thank you for checking it out

**Pull request merged**